core/foundation/inc/ROOT/StringUtils.hxx

```diff
@@ -23,6 +23,25 @@ namespace ROOT {
 
 std::vector<std::string> Split(std::string_view str, std::string_view delims, bool skipEmpty = false);
 
+std::string Join(const std::string& sep, const std::vector<std::string>& strings);
+
+/**
+ * \brief Concatenate several std::string
+ * \param[in] sep separator inbetween the strings
+ * \param[in] strings variadic list of strings
+ * \return the sep-delimited concatenation of strings
+ * \see https://stackoverflow.com/a/71983949/7471760
+ */
+template<class...T>
+inline std::string Join(const std::string& sep, T&&...strings) {
```

Comment:
What's exactly the usecase of this variadic template method? It's quite restrictive, because the length needs to be known at runtime. Also, the code is hard the read. If you have for example:
```c++
ROOT::Join("foo", "bar", "geek");
```
I find it hard to guess which of the args is the delimiter, the first or the last? I'd have to look up the signature. The `ROOT::Join("foo", {"bar", "geek"})` version would be more intuitive.

Also, I think the variadic template version would be redundant once we have the templated collection version I suggested, because then also then you can also just that one with an initializer list and it will also have no overhead.